plugins/security/nexus-ssl-plugin/src/main/java/com/sonatype/nexus/ssl/plugin/internal/ui/TrustStoreComponent.groovy

```
@@ -75,9 +101,65 @@ extends DirectComponentSupport
+    try {
+      trustStore.removeTrustCertificate(id)
+    }
+    catch (KeyNotFoundException e) {
+      throw new Exception("Certificate with fingerprint '${id}' not found")
+    }
+  }
+
+  static CertificateXO asCertificateXO(final Certificate certificate, final boolean inNexusSSLTrustStore)
+  throws Exception
+  {
+    String fingerprint = calculateFingerprint(certificate)
+
+    if (certificate instanceof X509Certificate) {
+      X509Certificate x509Certificate = (X509Certificate) certificate
+
+      Map<String, String> subjectRdns = getRdns(x509Certificate.subjectX500Principal.name)
+      Map<String, String> issuerRdns = getRdns(x509Certificate.issuerX500Principal.name)
+
+      return new CertificateXO(
+          id: fingerprint,
+          pem: CertificateUtil.serializeCertificateInPEM(certificate),
+          fingerprint: fingerprint,
+          serialNumber: x509Certificate.getSerialNumber().toString(),
+          subjectCommonName: subjectRdns.get("CN"),
+          subjectOrganization: subjectRdns.get("O"),
+          subjectOrganizationalUnit: subjectRdns.get("OU"),
+          issuerCommonName: issuerRdns.get("CN"),
+          issuerOrganization: issuerRdns.get("O"),
+          issuerOrganizationalUnit: issuerRdns.get("OU"),
+          issuedOn: x509Certificate.getNotBefore().getTime(),
+          expiresOn: x509Certificate.getNotAfter().getTime(),
+          inNexusSSLTrustStore: inNexusSSLTrustStore
+      )
+    }
+    else {
+      return new CertificateXO(
+          id: fingerprint,
+          pem: CertificateUtil.serializeCertificateInPEM(certificate),
+          fingerprint: fingerprint
+      )
+    }
+  }
+
+  static Map<String, String> getRdns(final String dn) {
+    Map<String, String> rdns = [:]
+    try {
+      LdapName ldapName = new LdapName(dn)
+      for (Rdn rdn : ldapName.rdns) {
+        rdns.put(rdn.type, rdn.value.toString())
+      }
+    }
+    catch (InvalidNameException e) {
+      // TODO ?
```

avoid eating exception